lib/postgrex/builtins.ex

```diff
@@ -161,3 +161,87 @@ defmodule Postgrex.Point do
     x: nil,
     y: nil]
 end
+
+defmodule Postgrex.Polygon do
+  @moduledoc """
+  Struct for Postgres polygon.
+
+  ## Fields
+    * `vertices`
+  """
+  @type t :: %__MODULE__{vertices: [Postgrex.Point.t]}
+
+  defstruct vertices: []
+end
+
+defmodule Postgrex.Line do
+  @moduledoc """
+  Struct for Postgres line.
+
+  Note, lines are stored in Postgres in the form `{a,b,c}`, which
+  parameterizes a line as `a*x + b*y + c = 0`.
+
+  ## Fields
+    * `a`
+    * `b`
+    * `c`
+  """
+  @type t :: %__MODULE__{a: float, b: float, c: float}
+
+  defstruct a: nil, b: nil, c: nil
+end
+
+defmodule Postgrex.LineSegment do
+  @moduledoc """
+  Struct for Postgres line segment.
+
+  ## Fields
+    * `point1`
+    * `point2`
+  """
+  @type t :: %__MODULE__{point1: Postgrex.Point.t, point2: Postgrex.Point.t}
+
+  defstruct point1: %Postgrex.Point{}, point2: %Postgrex.Point{}
```

Comment:
I don't think we should default to a point here since points default to `%Point{x: nil, y: nil}`. I don't think it makes sense to have a point with nil coordinates. The rest of the file also seems to default to nil for its fields.